lib/dry/cli/command.rb

```diff
@@ -316,13 +350,17 @@ def self.argument(name, options = {})
       #   # Options:
       #   #   --port=VALUE, -p VALUE
       def self.option(name, options = {})
-        @options << Option.new(name, options)
+        @_mutex.synchronize do
+          @options << Option.new(name, options)
+        end
       end
 
       # @since 0.1.0
       # @api private
       def self.params
-        (@arguments + @options).uniq
+        @_mutex.synchronize do
+          (@arguments + @options).uniq
```

Comment:
No mutations happening here, I don't think using a mutex is necessary?

Reply:
It's not about mutation, it's about consistent reading. Unserialized access can lead to anomalies. From the caller perspective, you anticipate `params` returns a snapshot of consistent data. When you access `@arguments` and `@options` without coordination the result may be inconsistent. Do you see the point? I may draw a diagram for better understanding.

Reply:
Got you, so basically, the mutex needs to synchronize around the reads and writes?

Reply:
Yes